src/lib.rs

```diff
@@ -432,6 +432,18 @@ pub mod actions {
         AddOverflowU(Bits),
         AddFp(Bits),
         And(Bits),
+        Asl(Bits),                    // Arithmetic shift left
+        AslOverflow(Bits),            // Overflag flag as a result of arithmetic shift left
```

Comment:
This is confusing as `AslOverflow`, despite having the more obvious name than `Carry`, is unaffected for most shifts. It's only affected on 1-bit shifts. Additionally, its actual value is pretty inscrutable and not particularly useful. `Carry` is almost always what you really want. I've tried to avoid the "overflow" vs "carry" terminology in the definition of actions, since it's heavily tied to the semantics of individual x86 instructions and we're trying to define CISC instructions in terms of atomic operations, ideally atomic operations that exist on multiple platforms. I would have the actions `ShiftROverflowU` and `ShiftROverflowS`, and `ShiftLOverflow`, which would be the value written to the carry flag, and I'd just use `Undefine` (or whatever we called it) for the overflow flag, since it's pretty non-trivial to use it correctly and I don't think we'd have any use for it. Admittedly the fact that the action called `Overflow` is written to the carry flag is a little unintuitive, but we've already used that nomenclature for `Add`/`Sub`/etc.

Reply:
Question on this " would have the actions ShiftROverflowU and ShiftROverflowS, and ShiftLOverflow, which would be the value written to the carry flag, and I'd just use Undefine (or whatever we called it) for the overflow flag, since it's pretty non-trivial to use it correctly and I don't think we'd have any use for it.". I am confused how ShiftROverflowU and ShiftROverflowS can be written to the carry flag on the same action?

Reply:
Ah, I just realised that I got this wrong. It should be `ShiftRUnderflowS`, `ShiftRUnderflowU` and and `ShiftLOverflow`, each representing the CF flag of `SAR`, `SHR` and `SAL`/`SHL` respectively. Here's an example of what I mean by that:

```rust
.instr("shr r32, cl", |new| {
    let left = new.param(INT_REG);
    let right = new.param(&regs::RCX);

    new.action_into(left, G::ShiftRU, [left, right]);
    new.action_into(&regs::CF, G::ShiftRUnderflowU, [left, right]);
})
.instr("sar r32, cl", |new| {
    let left = new.param(INT_REG);
    let right = new.param(&regs::RCX);

    new.action_into(left, G::ShiftRS, [left, right]);
    new.action_into(&regs::CF, G::ShiftRUnderflowS, [left, right]);
})
.instr("shl r32, cl", |new| {
    let left = new.param(INT_REG);
    let right = new.param(&regs::RCX);

    new.action_into(left, G::ShiftL, [left, right]);
    new.action_into(&regs::CF, G::ShiftLOverflow, [left, right]);
})
```